### PR TITLE
cli: disable automatic version checking by default

### PR DESCRIPTION
**Pull request by beardypig**
Adds a new option `--auto-version-check=(yes|no)` and suppresses the existing option `--no-version-check`. `--auto-version-check` defaults to `no`, and can re-enabled by setting `--auto-version-check=yes`.

See discussion in #590.

### CODE REVIEW

**Comment by gravyboat** [comment]
Awesome thanks @beardypig. I'm going to leave this open in case anyone else wants to weigh in, if not I'll merge it.

**Comment by karlo2105** [comment]
@beardypig 
Does it work like old switch in streamlinkrc config file?

**Comment by beardypig** [comment]
What you mean? Any option can go in the config file. In this case to turn the automatic check back in you need to set 
```
auto-version-check=yes
```

The existing `no-version-check` option is effective ignored.

**Comment by bastimeyer** [comment]
As a tiny improvement, I think the value should get implicitly set to "yes" when the `--auto-version-check` parameter is being used but without any values. This doesn't affect its default value of course. Maybe `true`, `false` and `1`, `0` could also be accepted as possible values.

I'm not sure how I feel about completely ignoring `--no-version-check`. I'm using this parameter in the Twitch GUI and it can't be disabled by the user. Adding the new `--auto-version-check` parameter and disabling the old one means that I need to bump the min required Streamlink version and change the parameters.

IMO, the `--no-version-check` should still disable the version check logic when set, regardless the value of `--auto-version-check`. Just for the sake of backwards compatibility, even though the default value of the new one is set to `no`.

**Comment by beardypig** [comment]
`--no-version-check` could be used as an override and always disable the version check regardless. Having an option with an implicit value is a bit troublesome, they can lead to unexpected errors and mean that some options might now work next to other ones.
It's easy to make it accept 1, 0, true, and false as well.

File: src/streamlink_cli/argparser.py [hunk 1]
```diff
@@ -261,10 +269,14 @@ def keyvalue(value):
     """
 )
 general.add_argument(
-    "--no-version-check",
-    action="store_true",
+    "--auto-version-check",
+    type=boolean,
+    metavar="{yes,true,1,on,no,false,0,off}",
+    default=False,
```
Comment (cdrage) [inline src/streamlink_cli/argparser.py]: By default should this be true? (having streamlink check the version on each usage by default)

Comment (beardypig) [inline src/streamlink_cli/argparser.py]: It's to disable that, see #590 :-)

Comment (cdrage) [inline src/streamlink_cli/argparser.py]: Ah poo. That's why. i agree about speeding it up :+1:

**Comment by gravyboat** [comment]
Merging this in, thanks @beardypig, with this change we can close https://github.com/streamlink/streamlink/pull/611 I believe unless anyone has other concerns.

**Pull request merged**